hera_cal/utils.py

```diff
@@ -1033,6 +1033,87 @@ def gain_relative_difference(old_gains, new_gains, flags, denom=None):
     return relative_diff, avg_relative_diff
 
 
+def red_average(hd, reds=None, bl_tol=1.0, wgt_by_int=False, inplace=False):
+    """
+    Redundantly average visibilities in a HERAData or UVData object.
+
+    Args:
+        hd : HERAData or UVData object
+            A UVData subclass object to redundantly average
+        reds : list, optional
+            Nested lists of antpair tuples to redundantly average.
+            E.g. [ [(1, 2), (2, 3)], [(1, 3), (2, 4)], ...]
+            If None, will calculate these from the metadata
+        bl_tol : float
+            Baseline redundancy tolerance in meters. Only used if reds is None.
+        wgt_by_int : bool
```

Comment:
Ah I see: in my mind this is the more conservative thing, so that's what should be default :) I guess its just a difference in perspective. I can make it default if this is a key issue for you, otherwise I'll leave it as is

Reply:
It's certainly my preference, but I'm not sure if it's a "key issue." If you provide a way to manually supply wgts, then I'm fine leaving this `False` by default.

Reply:
Okay, my preference is to leave it uniform by default but to make an issue discussing how to incorporate non-binary weights into this average (see response below). This is actually somewhat of a research question, as we need to calculate what the propagated Tsys of the averaged data will be under non-uniform weighting~I think I know what the answer is, but I haven't completely convinced myself.

Reply:
Actually thinking again, I think this is important to have, because it reflects what is done in redundant calibration--I forgot that we switched to weighting the data by the auto noise variance in the redcal solutions. I'll find a way to add this in.